test/e2e/pkg/client/auth_providers.go

```diff
@@ -0,0 +1,25 @@
+/*
```

Comment:
I see quite some files from pkg/client are copied from production to E2E test. Do we need to copy all of them in case some files are not changed in E2E at all?
Alternatively, we can copy the files that need to be changed for E2E, and still refer to the shared ones in the production?

Reply:
Discussed with @danfengliu, there are two unchanged files only, we will keep monitoring it, if more sharing files are required, we come back to change it together.